### PR TITLE
New toolbar cookie + Secure session/csrf cookies

### PR DESCRIPTION
**Pull request by mariusandra**
## Changes

This PR does two things.

First,

The [django documentation](https://docs.djangoproject.com/en/3.0/topics/security/#ssl-https) says the following:

> If a browser connects initially via HTTP, which is the default for most browsers, it is possible for existing cookies to be leaked. For this reason, you **should set** your `SESSION_COOKIE_SECURE` and `CSRF_COOKIE_SECURE` settings to True.

This PR does that. These settings now default to `True` and revert to `False` if running under DEBUG/TEST or if the env var `DISABLE_SECURE_COOKIES` is set.

This PR will be breaking if you're running posthog under regular HTTP in production (no nginx ssl proxy, no nothing). Then you need to set the `DISABLE_SECURE_COOKIES` flag.

This should be documented if the PR is merged.

Second,

This PR sets a new cross-domain cookie, `phtoolbar`, which is available to the `/decide` endpoint. It uses it to determine that the user is logged in and returns `true` for isAuthenticated... causing the toolbar to appear and ask for authorization.

This is a lot safer than setting the session cookie to be cross domain, as in case there's an error with this cookie (strong security in the browser, etc), nothing bad will happen. You'll just need to click "Launch Toolbar" manually.

Caveats: I'm not sure if I should add tests for this. Probably, if so, it's a WIP. I just ran out of time now.

## Checklist

- [ ] All querysets/queries filter by Team (if this PR affects any querysets/queries)
- [ ] Backend tests (if this PR affects the backend)
- [ ] Cypress E2E tests (if this PR affects the front and/or backend)

### CODE REVIEW

**Comment by mariusandra** [comment]
Tests are failing, setting to WIP/draft

**Comment by mariusandra** [comment]
This is waiting behind a new `posthog-js` release, which is waiting for a review on https://github.com/PostHog/posthog-js/pull/71 .

**Comment by mariusandra** [comment]
Small change from the text above. Instead of the `DISABLE_*` env, there's now a variable `SECURE_COOKIES` that should either be unset (defaults to `False` in DEBUG/TEST and `True` otherwise) or then explicitly set to True/False depending if the app is running behind https or not. This adds the `Secure` flag to all cookies, bringing us in line with what the django docs and Chrome's errors suggest we do.

This might cause trouble for people upgrading who have some special configs... and this new variable should be well documented.

---

Some issue still with cypress tests...?

File: posthog/middleware.py [hunk 1]
```diff
@@ -57,3 +57,36 @@ def __call__(self, request: HttpRequest):
         return HttpResponse(
             "Your IP is not allowed. Check your ALLOWED_IP_BLOCKS settings. If you are behind a proxy, you need to set TRUSTED_PROXIES. See https://posthog.com/docs/deployment/running-behind-proxy"
         )
+
+
+class ToolbarCookieMiddleware(SessionMiddleware):
```
Comment (timgl) [inline posthog/middleware.py:62]: In favour of a test for this

File: posthog/settings.py [hunk 2]
```diff
@@ -66,23 +66,31 @@ def print_warning(warning_lines: Sequence[str]):
...
             dsn=os.environ["SENTRY_DSN"], integrations=[DjangoIntegration()], request_bodies="always",
         )
 
-if get_bool_from_env("LOCAL_HTTPS", False):
-    SECURE_SSL_REDIRECT = False
-    SESSION_COOKIE_SECURE = True
-
 if get_bool_from_env("DISABLE_SECURE_SSL_REDIRECT", False):
     SECURE_SSL_REDIRECT = False
-    SESSION_COOKIE_SECURE = False
```
Comment (timgl) [inline posthog/settings.py:85]: is this correct? https://posthog.com/docs/deployment/running-behind-proxy#infinite-redirect

Comment (mariusandra) [reply posthog/settings.py:85]: That's correct. Explicitly setting `SESSION_COOKIE_SECURE` to `False` caused problems with logging in for users who had the `DISABLE_SECURE_SSL_REDIRECT` set. They were still behind SSL, but the redirect was just done via nginx. Thus the cookies had to be secure. 

Worse, in the last version of settings.py, this `SESSION_COOKIE_SECURE = False` was set in a way that it couldn't be overridden.

Comment (timgl) [reply posthog/settings.py:85]: Great!

**Comment by mariusandra** [comment]
This might be outside the scope for this PR, but after giving it a though, I'd actually nuke `DISABLE_SECURE_SSL_REDIRECT`, always set `SECURE_SSL_REDIRECT` to `False` and let users manage their own SSL.

This setting makes sense only for users who direct both their HTTP and HTTPS traffic directly to their app. How many people do that? What's the risk here vs benefit of an easier setup with fewer compilcated env variables?

**Comment by timgl** [comment]
The big one is Heroku, where if you don't set it you allow both http and https access.

**Comment by mariusandra** [comment]
- I added a test for the toolbar cookie middleware
- Cypress tests pass again. 
- This PR also now contains `cypress-terminal-report`, which gives detailed logs for cypress runs [if they fail](https://github.com/PostHog/posthog/runs/968559800). It helped debug my issue (bad host on XHR) and seems like a practical thing to have around:

image

Regarding heroku... oh well. I think then it could make sense to eventually flip the SSL redirect toggle, so it must be explicitly enabled. There are bigger fish to fry now.

**Pull request merged**